backend/src/lib/compression.ts

```diff
@@ -7,9 +8,216 @@ import {
   createGunzip,
   createGzip,
   createInflate,
+  inflateRawSync,
+  deflateRawSync,
   Zlib,
 } from 'zlib'
 import { logger } from './logger'
+import { ServerSideEvent, WatchEvent } from './server-side-events'
+import { getEventDict } from '../routes/events'
+import { getAppDict, ICompressedResource, ITransformedResource } from '../routes/aggregators/applications'
+
+type Dictionary = {
+  arr: string[]
+  map: Record<string, string>
+  add: (key: string) => string
+  get: (inx: number) => string
+}
+
+export function createDictionary(): Dictionary {
+  const arr: string[] = []
+  const map: Record<string, string> = {}
+  const add = (key: string): string => {
+    if (!(key in map)) {
+      map[key] = `${arr.length}`
+      arr.push(key)
+    }
+    return map[key]
+  }
+  const get = (inx: number) => {
+    return arr[inx]
+  }
+  return {
+    arr,
+    map,
+    add,
+    get,
+  }
+}
+
+// keys that point to values we should filter out
+const filterKeys = new Set(['finalizers', 'flags'])
```

Comment:
We already have a `pruneResources` function that strips certain things. It looks to me like these key names would apply recursively, and who knows where "flags" might appear in resource YAML. I think it would be better to centralize any pruning of data and to be explicit about what exact field we are targetting.